solarforecastarbiter/datamodel.py

```diff
@@ -1230,6 +1233,15 @@ def __post_init__(self):
                 'One of observation_id OR aggregate_id must be set')
 
 
+def __check_plot_spec__(plot_spec):
+    """Ensure that the provided plot specification is a valid JSON object"""
+    try:
+        spec_dict = json.loads(plot_spec)
+        validate(instance=spec_dict, schema={'type': 'object'})
+    except (json.JSONDecodeError, ValidationError):
+        raise ValueError('Figure spec must be a valid json object.')
+
+
 @dataclass(frozen=True)
 class ReportFigure(BaseModel):
```

Comment:
`PlotlyReportFigure`? Doesn't seem like a library-agnostic class.

Reply:
I think I misunderstood earlier, are you opposed to having an empty `ReportFigure` parent class that gets sub classed by BokehReportFigure and PlotlyReportFigure? I think I mistook your opposition to one class with optional fields as being opposed to that as well.

Reply:
> are you opposed to having an empty ReportFigure parent class that gets sub classed by BokehReportFigure and PlotlyReportFigure? 

I am not opposed to that. It sounds reasonable. @alorenzo175 should comment before I waste more of your time.

Reply:
sub-classes are ok, but we should make sure they include a ``type`` key so that it's easier to load into ``BokehReportFigure`` or ``PlotlyReportFigure`` when needed